tests/imfile-endmsg.regex-vg.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+# This is part of the rsyslog testbench, licensed under ASL 2.0
+# This test tests imfile endmsg.regex.
+USE_VALGRIND=true
+echo ======================================================================
+echo [imfile-endmsg.regex.sh]
+. $srcdir/diag.sh check-inotify
+. $srcdir/diag.sh init
+generate_conf
+add_conf '
+module(load="../plugins/imfile/.libs/imfile")
+
+input(type="imfile"
+      File="./'$RSYSLOG_DYNNAME'.crio.input"
+      Tag="file:" addMetadata="on" escapelf="on"
+      endmsg.regex="^[^ ]+ (stdout|stderr) F ")
+
+template(name="outfmt" type="list") {
+  property(name="$!all-json-plain")
+  constant(value=" ")
+  property(name="msg" format="json")
+  constant(value="\n")
+}
+
+if $msg contains "msgnum:" then
+ action(
+   type="omfile"
+   file=`echo $RSYSLOG_OUT_LOG`
+   template="outfmt"
+ )
+'
+if [ "x${USE_VALGRIND:-false}" == "xtrue" ] ; then
```

Comment:
I will merge this as is, but together with the constant "use_valgrind=true" this is against the testbench plumbing. 

We prefer to have dedicated valgrind tests, which always use valgrind, and those that do not use valgrind. The latter ones can also be run under ASAN control. ASAN and valgrind actually have a set of issue that they do NOT catch both, so running under both makes sense. Also, valgrind considerably changes multithreading properties, so many potential races do not occur when running under valgrind.

We could also change the testbench plumbing so that we do valgrind conditionally. But then we would also need to find a way to run the same test with and without valgrind. I am open to that, and happy to discuss options.

This is not really PR specific, and so I will merge without this being resolved. I would still think it makes sense to discuss the topic.